### PR TITLE
Master -> Control Plane

### LINKED ISSUE #688: cleanup: Master -> Control Plane
this is a request for community members to perform a cleanup on the word "master" in the Cluster API code base and documentation. Possibly send multiple PRs if needed.

As per the Kubernetes Steering committee the word "master" should be omitted as a user exposed term (UX) and instead the term to use should be "control plane".

It might be OK to keep the word "master" in function names and variables that are not API, as this would be hidden from the user, but ideally the change should be applied everywhere.

a github search reveals:
https://github.com/kubernetes-sigs/cluster-api/search?utf8=%E2%9C%93&q=master&type=

/kind cleanup
/help

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #688

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/crds/cluster_v1alpha1_machine.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ spec:
               properties:
                 controlPlane:
                   description: Semantic version of the Kubernetes control plane to
-                    run. This should only be populated when the machine is a master.
+                    run. This should only be populated when the machine is a controlplane.
```
Comment (roberthbailey) [inline config/crds/cluster_v1alpha1_machine.yaml]: If these aren't auto-generated, then they should all read:

"when the machine is intended to host the cluster's control plane."

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 2]
```diff
@@ -225,15 +225,15 @@ func (d *ClusterDeployer) updateClusterEndpoint(client clusterclient.Client, pro
 	// Update cluster endpoint. Needed till this logic moves into cluster controller.
 	// TODO: https://github.com/kubernetes-sigs/cluster-api/issues/158
 	// Fetch fresh objects.
-	cluster, master, _, err := getClusterAPIObject(client, clusterName, namespace)
+	cluster, controlplane, _, err := getClusterAPIObject(client, clusterName, namespace)
 	if err != nil {
 		return err
 	}
-	masterIP, err := provider.GetIP(cluster, master)
+	controlplaneIP, err := provider.GetIP(cluster, controlplane)
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: controlPlaneIP or clusterEndpoint or clusterIP. I lean towards clusterEndpoint personally.

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 3]
```diff
@@ -225,15 +225,15 @@ func (d *ClusterDeployer) updateClusterEndpoint(client clusterclient.Client, pro
...
 	if err != nil {
-		return fmt.Errorf("unable to get master IP: %v", err)
+		return fmt.Errorf("unable to get controlplane IP: %v", err)
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: "unable to get the control plane IP" or "unable to get the IP of the control plane" or "unable to get the IP of the cluster endpoint"

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 4]
```diff
@@ -404,30 +404,30 @@ func getClusterAPIObject(client clusterclient.Client, clusterName, namespace str
 		return nil, nil, nil, fmt.Errorf("unable to fetch cluster %v in namespace %v: %v", clusterName, namespace, err)
 	}
 
-	master, nodes, err := extractMasterMachine(machines)
+	controlplane, nodes, err := extractControlPlaneMachine(machines)
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: controlPlaneMachine

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 5]
```diff
@@ -404,30 +404,30 @@ func getClusterAPIObject(client clusterclient.Client, clusterName, namespace str
...
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("unable to fetch master machine in cluster %v in namespace %v: %v", clusterName, namespace, err)
+		return nil, nil, nil, fmt.Errorf("unable to fetch controlplane machine in cluster %v in namespace %v: %v", clusterName, namespace, err)
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: control plane

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 6]
```diff
@@ -404,30 +404,30 @@ func getClusterAPIObject(client clusterclient.Client, clusterName, namespace str
...
 	}
-	return cluster, master, nodes, nil
+	return cluster, controlplane, nodes, nil
 }
 
-// extractMasterMachine separates the master (singular) from the incoming machines.
+// extractcontrolplaneMachine separates the controlplane (singular) from the incoming machines.
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: "... separates the machine running the control plane (singular) ..."

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 7]
```diff
@@ -404,30 +404,30 @@ func getClusterAPIObject(client clusterclient.Client, clusterName, namespace str
...
 // This is currently done by looking at which machine specifies the control plane version
-// (which implies that it is a master). This should be cleaned up in the future.
-func extractMasterMachine(machines []*clusterv1.Machine) (*clusterv1.Machine, []*clusterv1.Machine, error) {
+// (which implies that it is a controlplane). This should be cleaned up in the future.
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: I'd just delete the parenthetical; now that the text uses control plane I don't think it adds any value.

File: cmd/clusterctl/clusterdeployer/clusterdeployer.go [hunk 8]
```diff
@@ -404,30 +404,30 @@ func getClusterAPIObject(client clusterclient.Client, clusterName, namespace str
...
+func extractControlPlaneMachine(machines []*clusterv1.Machine) (*clusterv1.Machine, []*clusterv1.Machine, error) {
 	nodes := []*clusterv1.Machine{}
-	masters := []*clusterv1.Machine{}
+	controlplanes := []*clusterv1.Machine{}
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer.go]: controlPlaneMachines

File: cmd/clusterctl/clusterdeployer/clusterdeployer_test.go [hunk 9]
```diff
@@ -557,7 +557,7 @@ func TestClusterCreate(t *testing.T) {
 			expectErr:                           true,
 		},
 		{
-			name:                                "fail create master",
+			name:                                "fail create controlplane",
```
Comment (roberthbailey) [inline cmd/clusterctl/clusterdeployer/clusterdeployer_test.go]: control plane machine

**Comment by vincepri** [comment]
@roberthbailey everything should be fixed, I also did a quick search for other instances of the same

File: docs/book/provider_implementations/create_actuators.md [hunk 10]
```diff
@@ -153,9 +153,9 @@ func (a *Actuator) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine)
         return "", fmt.Errorf("TODO: Not yet implemented")
 }
 
-// GetKubeConfig gets a kubeconfig from the master.
-func (a *Actuator) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.Machine) (string, error) {
-        log.Printf("Getting IP of machine %v for cluster %v.", master.Name, cluster.Name)
+// GetKubeConfig gets a kubeconfig from the running the control plane.
```
Comment (neolit123) [inline docs/book/provider_implementations/create_actuators.md]: `running control plane` ?

File: pkg/controller/machine/controller.go [hunk 11]
```diff
@@ -193,6 +193,35 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileMachine) create(ctx context.Context, machine *clusterv1.Machine) error {
```
Comment (roberthbailey) [inline pkg/controller/machine/controller.go]: These functions were just removed. You should just remove this file from your commit.

Comment (vincepri) [reply pkg/controller/machine/controller.go]: Whoops, fixing that!

**Pull request merged**